### DISCUSSION TITLE
osg-satellite

### DISCUSSION THREAD

**Discussion by naive1233423** [omnetpp/omnetpp#1274] [Q&A]
I am trying to establish a communication project between satellites and ground stations based on the OSG satellite model, but during my operation, there may be instances where the satellite does not orbit the Earth while communicating with the ground station. Please give me some advice, thank you

**Comment by rhornig** [comment]
The OSG sample is about the visualization of 3D graphics. It does not have any satellite related models. I.e. no mobility models describing the satellite movement and no propagation models describing the line of sight changes between satellites. IMHO, this is not a good starting point for a satellite model (apart that it looks nice). I would take a look at the OSS3 model, that my give you a mobility model

**Reply by naive1233423** [reply]
Thank you for your answer. I have seen the code running result of OS3, but it is a 2D demonstration, not a 3D model. The research I am currently conducting aims to achieve 3D demonstrations
